src/Analyser/MutatingScope.php

```diff
@@ -1419,6 +1424,12 @@ private function resolveType(Expr $node): Type
 
 			$resultType = TypeCombinator::union($leftNumberType, $rightNumberType);
 			if ($node instanceof Expr\AssignOp\Div || $node instanceof Expr\BinaryOp\Div) {
+				// division on strings is not allowed. its allowed on numeric strings though.
+				if ($leftType->isString()->yes() && !$leftType->isNumericString()->yes() && !$leftType->isLiteralString()->yes()
+					|| $rightType->isString()->yes() && !$rightType->isNumericString()->yes() && !$rightType->isLiteralString()->yes()) {
```

Comment:
1) `||` vs. `&&` operator priority needs to be disambiguated with parentheses
2) I don't know why literal-string is mentioned in this condition

Reply:
1. fixed
2. I mixed up my mental model in "123" would be a literal-string, but it is also a numeric-string, so the additional check is not necessary. fixed a bug so we don't accept `literal-string` for binary operations